pallets/clusters/src/lib.rs

```diff
@@ -221,26 +237,58 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			role_hash: Hash256,
 			cluster_hash: Hash256,
-			new_settings: RoleSettings,
+			new_members_list: Option<Vec<Hash256>>,
+			new_settings: Option<RoleSettings>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			if new_settings.data.is_empty() || new_settings.name.is_empty() {
-				return Err(Error::<T>::InvalidInputs.into());
+			if new_members_list.is_none() && new_settings.is_none() {
+				return Err(Error::<T>::InvalidInputs.into())
 			}
 
-			// only the owner of the cluster can do this operation
+			// Check that the caller is the owner of the cluster
 			let cluster = <Clusters<T>>::get(&cluster_hash).ok_or(Error::<T>::SystematicFailure)?;
 			ensure!(who == cluster.owner, Error::<T>::NoPermission);
 
+			// Check that the role exists in the cluster and in storage
 			let roles_in_cluster =
 				<Clusters<T>>::get(&cluster_hash).ok_or(Error::<T>::ClusterNotFound)?.roles;
 			ensure!(roles_in_cluster.contains(&role_hash), Error::<T>::RoleNotFound);
 			ensure!(<Roles<T>>::contains_key(role_hash), Error::<T>::RoleNotFound);
 
+			let current_settings =
+				<Roles<T>>::get(role_hash).ok_or(Error::<T>::SystematicFailure)?.settings;
+
+			// If there are no new settings, override. Otherwise use the old ones.
+			let settings = if let Some(new_settings) = new_settings {
+				if !new_settings.data.is_empty() && !new_settings.name.is_empty() {
+					new_settings
```

Comment:
can you batch remove settings?

Reply:
Ok. So now if the cluster's owner sends an empty list of role settings, it will [delete the existing ones](https://github.com/fragcolor-xyz/clamor/blob/8276f14d8e8d6eb42a1fcc163adbc348619888ed/pallets/clusters/src/lib.rs#L302).
Same in case the list of [members passed as input is empty](https://github.com/fragcolor-xyz/clamor/blob/8276f14d8e8d6eb42a1fcc163adbc348619888ed/pallets/clusters/src/lib.rs#L291).